crypto/evp/bio_md.c

```diff
@@ -140,6 +140,12 @@ static long md_ctrl(BIO *b, int cmd, long num, void *ptr)
     ctx = BIO_get_data(b);
     next = BIO_next(b);
 
+    // add check against NULL
+    if(!ctx)
+    {
+        return 0;
+    }
+
```

Comment:
This can happen only in case of memory corruption as the data is always set in md_new. I do not think it is worth adding such check.

Reply:
emm... yes, I just think adding this check can handle this error , preventing program crash.

Reply:
It appears it might also happen if someone calls BIO_set_md_ctx on an md bio with a NULL pointer (though why someone would do so is a mystery)

Reply:
@icy17, no, it cannot handle memory corruption.  Even if it could, it would be better to crash because nothing can be trusted after memory corruption.  Literally anything could happen.